app/scripts/migrations/160.ts

```diff
@@ -201,11 +203,25 @@ function transformState(state: Record<string, unknown>) {
 
     updatedSubjects.push(key);
 
+    const existingCaip25Caveat = getExistingCaip25PermissionCaveat(subject);
     const caip25PermissionCaveatWithCurrentChainIdSet = addPermittedEthChainId(
-      getExistingCaip25PermissionCaveat(subject),
+      existingCaip25Caveat,
       currentChainId,
     );
 
+    let caip25PermissionCaveatWithCurrentChainIdAndAccountsSet =
+      caip25PermissionCaveatWithCurrentChainIdSet;
+
+    if (existingCaip25Caveat) {
+      const alreadyPermissionedAccounts = getEthAccounts(existingCaip25Caveat);
+      if (alreadyPermissionedAccounts) {
```

Comment:
empty array is always truthy

Reply:
should we leave a comment what this block is trying to achieve? or nah because specs help clarify?